### DISCUSSION TITLE
Play System update status?

### DISCUSSION THREAD

**Discussion by stevelitchfield** [Archfx/duo-de#117] [General]
Hi all

I'm enjoying playing with, and even using, Duo-de etc. See my YouTube videos on it, for example. But when I check Play System status, it says '15', with no date etc. Given that the Play APIs and modules are rather critical to things working properly, can I ask whether this could be clarified? 

And do we inherit newer Play System modules when we update via PhhTreble etc?

Just wondered! Thanks for all you do!

**Comment by Archfx** [comment]
Hello @stevelitchfield. I appreciate your content and I think it improved the awareness about DUO-DE among enthusiasts.

By  `Play System update` did you mean the system updates (OTA)?

**Reply by stevelitchfield** [reply]
No, I'm talking about the Play System middleware that is updated by Google semi-regularly for every Android device. In a typical phone, the status of these is reported alongside the traditional Android kernel 'Security update' status.

For example, my Sony Xperia might have October 2024 Android security but July 2024 Play System update.

So... when you update the Duo-DE ROM, does this include the Play System/Services middleware, or does the modded phone still need to grab updates to these from Google directly?

**Reply by Archfx** [reply]
I see, thanks for clarifying! Some of this information isn't properly displayed in the settings because we're still on the A15 beta releases. For example, the build number was broken until a very recent DUO-DE update, even though the system is functioning correctly internally. The Play System update status might be similarly affected. We have to wait until stable releases are available from Google for the public.

**Reply by stevelitchfield** [reply]
Ok thanks, I’ll be patient! 

I do have more videos planned on Duo-DE, make sure you’re subscribed to stevelitchfield Shorts on YouTube etc! 8-)

**Reply by Archfx** [reply]
That sounds awesome! Looking forward to what you have planned!

**Comment by stevelitchfield** [comment]
Just did https://www.youtube.com/shorts/Cf5Q3VVWXW4 too 8-)

After I shot this, I've hit a minor glitch not mentioned in your forums. I
was fiddling with folding it each way to get Pixel Camera on the left
screen and the homescreens completely reset themselves. Even after a
restart now, they're empty and with default dock icons.

Has my set-up gone for good or is there a cache somewhere I need to wipe to
restore my widgets and placements?

Many thanks

**Comment by stevelitchfield** [comment]
Actually, it was a mere 5 mins to set things up again on the homescreens.
But I still thought you should know of this behaviour!

PS. 'Pixel Camera' in the Play Store (from Google LLC) isn't fully featured
on the Duo but still produces excellent results.

**Reply by Archfx** [reply]
Some unhandled exception caused the launcher to discard the cache. It would be helpful if you can reproduce this while collecting a log and posting it here.

You can collect the log using the following command

```shell
adb -d logcat | grep launcher3 > launcher3_log.txt
```

If we can identify the issue, we can directly fix it.

**Comment by stevelitchfield** [comment]
I don’t actually have adb handy on a PC. Mike Warner (famous flasher) did
the Duo-DE flash for me. Sorry.

I might have a PC in the loft. Anyway, will see what happens in future!!

Steve

(Sent while mobile, apologies for brevity!)

**Reply by Archfx** [reply]
No worries, may I know the build number of the DUO-DE?
This information is available in `settings->about device->build number`.

**Comment by stevelitchfield** [comment]
treble_arm64_bgN-userdebug 15 AP3A.241105.008 eng.aruna.20241129.203334

Steve

**Comment by stevelitchfield** [comment]
Oh, one request - and I haven't seen it in your forum. Or even know whether
it's possible.

At the moment, when you collapse both screens to one by folding the Duo
over, the Trubuchet laucher tries to combine the two. Is there a way to
turn on behaving like a Pixel Fold, so collapsing the Duo would show 'page
1', then opening is 'page 1' and 'page 2' and so on, etc?

Just thought I'd ask!! Thanks again!

**Reply by Archfx** [reply]
In the current builds, Launcher maintains two configurations (artifact placement) for the tablet mode and phone mode. You should be able to move things around in two modes separately. It switches back between those two (placement configurations) when you switch postures.

> PS: However, certain cases may not work as expected due to the complexity of switching between modes. I believe this is one of the reasons Microsoft didn't want to pursue the Duo after a few iterations.

**Comment by stevelitchfield** [comment]
Alas that’s not what I see. If I delete an icon or move it one layout then
it changes on the other - they’re not separate at all.

Odd!

Steve

(Sent while mobile, apologies for brevity!)
